storage/cleanup.go

```diff
@@ -194,6 +232,17 @@ func (m *cleanupManager) cleanupNamespaceFilesetFiles(earliestToRetain time.Time
 	return multiErr.FinalError()
 }
 
+func (m *cleanupManager) cleanupNamespaceSnapshotFiles(shards []databaseShard) error {
+	multiErr := xerrors.NewMultiError()
+	for _, shard := range shards {
+		if err := shard.CleanupSnapshots(); err != nil {
+			multiErr = multiErr.Add(err)
+		}
+	}
+
+	return multiErr.FinalError()
+}
+
```

Comment:
Doesn't the commit log cleanup logic need to change to account for the snapshot files?

Reply:
Agreed it does, but I will probably just do that in the P.R where I implement reads otherwise this P.R would leave M3DB in a "broken" state where if you had snapshotting on and were truncating commit logs, but still couldn't read the snapshots.

Also it will be a decent amount of code to implement / test this properly and the P.R is huge already

Reply:
Yeah completely fair. I like the way you have it right now - with the snapshot write path enabled as it is now, you can still run in prod without affecting durability guarantees.